src/lsm/lsm_tree.c

```diff
@@ -520,11 +521,11 @@ __wt_lsm_tree_get(WT_SESSION_IMPL *session,
 {
 	WT_DECL_RET;
 
-	WT_ASSERT(session, F_ISSET(session, WT_SESSION_LOCKED_HANDLE_LIST));
-
-	ret = __lsm_tree_find(session, uri, exclusive, treep);
+	WT_WITH_HANDLE_LIST_READ_LOCK(session,
+	    ret = __lsm_tree_find(session, uri, exclusive, treep));
 	if (ret == WT_NOTFOUND)
-		ret = __lsm_tree_open(session, uri, exclusive, treep);
+		WT_WITH_HANDLE_LIST_WRITE_LOCK(session,
```

Comment:
With this change, is it possible that multiple threads will try to open the tree sequentially? i.e: multiple threads could get WT_NOTFOUND returns from the ``__lsm_tree_find`` call, and all attempt to open the tree. Maybe they need to re-check find once the write lock is acquired.

Hmm - answering my own question - the ``__lsm_tree_open`` implementation already does that. Maybe adding a comment to that effect would be worthwhile.